core/src/com/unciv/models/ruleset/unique/UniqueParameterType.kt

```diff
@@ -15,6 +15,10 @@ enum class UniqueParameterType(val parameterName:String) {
             else null
         }
     },
+    NumberOrAll("amountOrAll") {
```

Comment:
Denied, this is a clear "amount". "Reveal up to All" throws human parse errors

Reply:
```json
    {
        "name": "reveal nearby Barbarian camps",
        "notification": "You find evidence of Barbarian activity. Nearby Barbarian camps are revealed!",
        "uniques": ["Reveal up to [All] [Barbarian encampment] within a [10] tile radius"]
    },
```
Not mine

Reply:
> throws human parse errors

:grin:  and the human throws his breakfast in to the exception bucket (Mr Creosote)

Reply:
Eeww yukkk
"amount/param" then, and please add a todo to that

Reply:
Wait - the "/" is parsed as alternate types? OK, forgot, that would obviously obviate the need for _that_ kludge type

Reply:
That is, this shouldn't be an enum value

Reply:
No "/" isn't parsed - didn't someone code that? Thought I saw it earlier. Do we want? Right in UniqueType.init?

Reply:
And that init could really use that safeValueOf generic I got in stash - time to unstash?
```kotlin
inline fun <reified T: Enum<T>> enumValueOfOrNull(name: String?): T? {
    if (name == null) return null
    return try {
        enumValueOf<T>(name)
    } catch (ex: IllegalArgumentException) {
        println("Could not parse\"$name\" as ${T::class.java.simpleName}.")
        null
    }
}
```
Each enum then can do a proxy easily as one-liner...

Reply:
Hey, that looks like code I've written :p
On second thought, it's probably best to split this out into two uniques?

Reply:
Our unit tests do not balk at parameter type violations as defined in UniqueParameterType.getErrorSeverity at all - uh, where were we? That still on the to-do list? Anyway, separating that all barb camp (plural is another readability problem) ruin reward - not this PR?

Reply:
OK, `checkModLinks` does not check the _ruin rewards_ and I was only looking for a test failure manipulation those... Yet another PR.

Reply:
Used a simpler safeValueOf in that latest push, also implements "/" for autoparsed placeholder params, so now those lists can have size>1